libcxx/test/std/algorithms/alg.sorting/alg.clamp/ranges.clamp.pass.cpp

```diff
@@ -82,31 +93,29 @@ constexpr bool test() {
     auto low = S{20};
     auto high = S{30};
     // Check that the value category of the projection return type is preserved.
-    assert(&std::ranges::clamp(val, low, high, Comp{}, &S::lvalue_proj) == &low);
```

Comment:
Can you say more about why these tests go away?

Reply:
This PR was lifted from https://github.com/llvm/llvm-project/pull/66315 which was closed unexpectedly, and the discussion over there explain it. Not replicating the full story here cause it's kind of complicated.

But long story short, the potential-double-moving is mandated by the standard (and if it is harmful then the projection is not equality-preserving, which means that the user did something bad -- it's still wacky that we can't diagnose it at compile-time). So these tests were in fact wrong from the start.

Reply:
The test above which does:

```
std::same_as<const int&> decltype(auto) ret = std::ranges::clamp(val, low, high);
assert(ret == 2);
assert(&ret == &val);
```

is the one that checks that we return a reference to the right object. This comparing of addresses was basically not necessary AFAICT.